### PR TITLE
[v1.9] Backport nix derivation for static builds

### PR DESCRIPTION
**Pull request by hswong3i**
Also see #6402

nix-based static binary could be found from:
- https://github.com/alvistack/libpod/releases/tag/v1.9.3
- https://github.com/alvistack/libpod/releases/tag/v2.0.0-rc2

### CODE REVIEW

**Comment by mheon** [comment]
You need to add a Signed-off-by line to your commit - `git commit --amend -s --no-edit` should work

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@baude @mheon @giuseppe @vrothberg @TomSweeneyRedHat @jwhonce @QiWang19 PTAL

**Comment by vrothberg** [comment]
I have no expertise in Nix and cannot review.

**Comment by hswong3i** [comment]
> I have no expertise in Nix and cannot review.

Could simply try with this:
```
curl -L https://nixos.org/nix/install | sh
git clone https://github.com/alvistack/libpod.git
cd libpod
git checkout v1.9-linux-amd64
nix build -f nix/
./result-bin/bin/podman --version
ls -lah ./result-bin/bin/podman
ldd ./result-bin/bin/podman
file ./result-bin/bin/podman
```

Example result:
```
$ ./result-bin/bin/podman --version
podman version 1.9.4-dev

$ ls -lah ./result-bin/bin/podman
-r-xr-xr-x 1 hswong3i hswong3i 50M Jan  1  1970 ./result-bin/bin/podman

$ ldd ./result-bin/bin/podman
    not a dynamic executable

$ file ./result-bin/bin/podman
./result-bin/bin/podman: ELF 64-bit LSB executable, x86-64, version 1 (GNU/Linux), statically linked, for GNU/Linux 2.6.32, Go BuildID=pVTlxXbP3zJwnIv270_g/n4422kj1lRjvCMVkckKC/h9rdwRGSToOL2cYUz5T-/hJIP5OfmT5Or4zlx42Y6, not stripped
```

**Comment by rhatdan** [comment]
@saschagrunert PTAL

File: .gitignore [hunk 1]
```diff
@@ -1,33 +1,34 @@
...
+contrib/spec/podman.spec
+*.coverprofile
 /docs/*.[158]
 /docs/*.[158].gz
-/docs/remote
 /docs/build/
+/docs/remote
+.gopathok
+.idea*
+.nfs*
```
Comment (saschagrunert) [inline .gitignore:16]: Let's put changes like this in dedicated PRs. Makes review much harder if it's not clear what has been added or removed in the first place :innocent:

Comment (hswong3i) [reply .gitignore]: Fixed

**Comment by hswong3i** [comment]
Compile now successful: https://github.com/alvistack/libpod/releases/download/v1.9.3/podman-v1.9.3-linux-amd64

```
$ ls -lah ./podman-v1.9.3-linux-amd64
-r-xr-xr-x 1 hswong3i hswong3i 44M May 31 11:38 ./podman-v1.9.3-linux-amd64
    
$ ldd ./podman-v1.9.3-linux-amd64
    not a dynamic executable 
 
$ file ./podman-v1.9.3-linux-amd64
./podman-v1.9.3-linux-amd64: ELF 64-bit LSB executable, x86-64, version 1 (GNU/Linux), statically linked, for GNU/Linux 2.6.32, Go BuildID=8aoQlir-hnIlUFgPjNpP/HSxtS1j80TAJXsk5b3Ly/BQf_LNRIsfb5jKfcQMNO/YygPNHhAHpdqlPmgvpOs, stripped

$ sudo ./podman-v1.9.3-linux-amd64 --version
podman-v1.9.3-linux-amd64 version 1.9.3
```

Therefore could focus on CI/CD fixes ;-)

**Comment by vrothberg** [comment]
Can you rebase the commits?

**Comment by hswong3i** [comment]
> Can you rebase the commits?

Checked and already rebased and updated with v1.9

**Comment by hswong3i** [comment]
Let's focus on master ;-)
